### PR TITLE
FIX: no error when double save post with poll

### PR DESCRIPTION
**Pull request by lis2**
Some plugins hook into Post after save to set custom fields and save again.

For example: https://github.com/discourse/discourse-category-experts/blob/main/lib/category_experts/post_handler.rb#L27

Problem is that in case like that `raw_changed?` is false but all callback are triggered. `extracted_polls` is class attribute therefore that should be reset with each attempt.

That was causing an error:
```
#<ActiveRecord::RecordNotUnique: PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint 
"index_polls_on_post_id_and_name" DETAIL:  Key (post_id, name)=(8967, poll) already exists.

```

### CODE REVIEW

File: plugins/poll/plugin.rb [hunk 1]
```diff
@@ -486,6 +486,7 @@ def groupable_user_fields
   end
 
   validate(:post, :validate_polls) do |force = nil|
+    self.extracted_polls = []
```
Comment (SamSaffron) [inline plugins/poll/plugin.rb:489]: this feels a bit weird. a side effect of calling validations is setting / resetting a var to []. 

Is there any other way we can do this fix? More nil checks elsewhere?

File: plugins/poll/plugin.rb [hunk 2]
```diff
@@ -467,6 +467,7 @@ def groupable_user_fields
 
       after_save do
         polls = self.extracted_polls
+        self.extracted_polls = []
```
Comment (SamSaffron) [inline plugins/poll/plugin.rb:470]: @ZogStriP are you happy with `after_save` doing the reset here?

Comment (ZogStriP) [reply plugins/poll/plugin.rb:470]: Yes, I think that's fine.

```suggestion
        self.extracted_polls = nil
```

`nil` or `{}` would be slightly more correct.

File: plugins/poll/spec/models/poll_spec.rb [hunk 3]
```diff
@@ -81,6 +81,11 @@
 
       expect(poll.post).to eq(post)
     end
+  end
 
+  it "is not throwing an error when double save" do
+    post = Fabricate(:post, raw: "[poll]\n- A\n- B\n[/poll]")
+    post.custom_fields["expert"] = true
```
Comment (ZogStriP) [inline plugins/poll/spec/models/poll_spec.rb:88]: Not sure how adding this line test for "double save" though. I _think_ it requires the `discourse-category-experts` plugin to be installed?

Comment (lis2) [reply plugins/poll/spec/models/poll_spec.rb]: You are right, it is not neccessary.

Without fix, just save! twice is raising an error:
```
 post = Fabricate(:post, raw: "[poll]\n- A\n- B\n[/poll]")
 post.save!
```

The reason for that is that raw is same and extraced_polls were not reset:
https://github.com/discourse/discourse/blob/master/plugins/poll/plugin.rb#L489

**Review by ZogStriP — APPROVED** [review]

**Pull request merged**